### PR TITLE
Reprovision s390x machines

### PR DESCRIPTION
**Pull request by AshCripps**
READY FOR REVIEW

This is to replace the current s390x machines with 5 machines in the new datacentre - 1 release, 4 test

### CODE REVIEW

**Comment by AshCripps** [comment]
Part of this PR is to update every reference from rhel72 to rhel7 because its just plain wrong otherwise, Let me know if ive missed anything or if thats likely to cause problems.

**Comment by richardlau** [comment]
> 
> 
> Part of this PR is to update every reference from rhel72 to rhel7 because its just plain wrong otherwise, Let me know if ive missed anything or if thats likely to cause problems.

We'd need to go through all of the jobs on Jenkins (both test and release) and update (which we'll have to do regardless if we're adding new machine names rather than replacing the existing ones).

**Comment by AshCripps** [comment]
all five machines are ansibled and this pr is ready to be reviewed

File: ansible/roles/jenkins-worker/tasks/main.yml [hunk 1]
```diff
@@ -123,31 +73,6 @@
...
-  lineinfile:
-    dest: /etc/sysconfig/iptables
-    insertafter: ":OUTPUT ACCEPT.*]"
-    line: "-A INPUT -s 127.0.0.2/32 -d 127.0.0.1/32 -j ACCEPT"
-  when: "'rhel72-s390x' in inventory_hostname"
-
-- name: Firewall | make the new firewall rules take effect
-  command: systemctl restart iptables
-  when: "'rhel72-s390x' in inventory_hostname"
-
```
Comment (richardlau) [inline ansible/roles/jenkins-worker/tasks/main.yml]: Are these no longer required? Or moved to a manual step?

Comment (AshCripps) [reply ansible/roles/jenkins-worker/tasks/main.yml]: No longer required I believe, These steps failed on the machines but make+test still pass. Need to try with a non master branch to double check

File: ansible/roles/baselayout/tasks/main.yml [hunk 2]
```diff
@@ -41,9 +41,9 @@
   when: os|startswith("smartos")
   shell: coreadm -i /home/iojs/cores/core.%f.%p -e process -d global
 
-- name: disable sftp
-  notify: restart sshd
-  lineinfile: state=absent dest="{{ ssh_config }}" regexp=^Subsystem(\s+)sftp
+#- name: disable sftp
```
Comment (sam-github) [inline ansible/roles/baselayout/tasks/main.yml]: 1. does sftp even exist on those machines?
2. if so, how to disable it?

File: ansible/roles/jenkins-worker/meta/main.yml [hunk 3]
```diff
@@ -2,7 +2,4 @@
 
 #
 # this role depends on java-base
-#
-
-dependencies:
-- { 'role': 'gcc', when: os|startswith("rhel7") }
+#
```
Comment (sam-github) [inline ansible/roles/jenkins-worker/meta/main.yml]: fix this

File: ansible/inventory.yml [hunk 4]
```diff
@@ -36,6 +36,7 @@ hosts:
 
     - ibm:
         aix71-ppc64_be-1: {ip: 129.33.196.199, user: b9s010a}
+        rhel7-s390x-1: {ip: 148.100.86.101, user: linux1}
```
Comment (sam-github) [inline ansible/inventory.yml]: @nodejs/build-infra Please add this IP to the Jenkins firewall.

File: ansible/inventory.yml [hunk 5]
```diff
@@ -98,6 +97,10 @@ hosts:
     - ibm:
         aix71-ppc64_be-1: {ip: 129.33.196.197, user: b9s010a}
         aix71-ppc64_be-2: {ip: 129.33.196.198, user: b9s010a}
+        rhel7-s390x-1: {ip: 148.100.86.21, user: linux1}
+        rhel7-s390x-2: {ip: 148.100.86.117, user: linux1}
+        rhel7-s390x-3: {ip: 148.100.86.28, user: linux1}
+        rhel7-s390x-4: {ip: 148.100.86.94, user: linux1}
```
Comment (sam-github) [inline ansible/inventory.yml]: @nodejs/build-infra Please add these 4 IPs to the Jenkins firewall.

**Comment by sam-github** [comment]
@richardlau / @nodejs/build-wg  Any reason for me not to squash merge this right now? It doesn't touch select-compiler.sh or the groovy files, just ansible, so I can't see it interfering with the ongoing security release, it looks entirely benign.

**Comment by richardlau** [comment]
> 
> 
> @richardlau / @nodejs/build-wg Any reason for me not to squash merge this right now? It doesn't touch select-compiler.sh or the groovy files, just ansible, so I can't see it interfering with the ongoing security release, it looks entirely benign.

No objections from me.

cc @nodejs/build

**Comment by mhdawson** [comment]
@sam-github IPs added to firewalls.

File: ansible/roles/jenkins-worker/tasks/main.yml (hunk 1 above)
Comment (richardlau) [reply ansible/roles/jenkins-worker/tasks/main.yml]: @AshCripps @sam-github I think these rules were for libuv: https://github.com/nodejs/build/issues/2080#issuecomment-567280766 and https://github.com/libuv/libuv/pull/2185#issuecomment-467290846

**Pull request merged**